Assets/UI/Panels/citypanel.lua

```diff
@@ -115,7 +115,23 @@ function CQUI_CityviewEnableManager()
 end
 
 -- ===========================================================================
-function CQUI_CityviewDisableManager()
+function CQUI_CityviewDisableManager(forceDisable:boolean)
```

Comment:
I am confused by the name forceDisable". If you call it with true then it actually doesn't go through extra steps and just does the same as previously. So, what is the point of adding stuff and calling it the way that the stuff is actually omitted?

Reply:
I hinted at this very briefly in my other response, but this variable was added so that I didn't need to make a separate function, though maybe that would be better for clarity, or maybe changing the name of the variable to "closeAll". The idea is that if it is true, it will close the City View entirely. So if you were previously in the Placement Mode, it will not return to the standard City View and instead close out of the view entirely. If forceDisable is not true, then it will change the interface to either CITY_MANAGEMENT or SELECTION based on whether Placement Mode is active or not. This lets the "CQUI_OnInterfaceModeChanged" function decide what to do, which is where a lot of the logic seems to be written currently. If Placement Mode is active, it will return to the standard City View. If it is not, it will disable the City View entirely.

This variable is pretty much only used by the CityPanel.lua file, though it is used once in CityBannerManager_CQUI.lua as well. Everywhere else, this variable isn't used and the new code is run.

Reply:
I definitely would go with separate function, just as your initial though was. Two totally separated execution paths = two functions.
Each function should have a clear and (best) singular purpose, and params should be related to the purpose. Making a function with 2 purposes and a param to distinguish between them, even if the purposes are related, it is rather bad coding practice.

Reply:
Understood! I separated the function into a separate one now that can now be called with `LuaEvents.CQUI_CityviewDisableCurrentMode()`. This function will, as the name says, disable the current city view mode. So if in the Placement mode, it will disable that mode and then restore the standard city view. Otherwise, it will disable the city view entirely.

Reply:
The code is more readable now!